sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala

```diff
@@ -95,9 +95,128 @@ object CodeFormatter {
     new CodeAndComment(code.result().trim(), map)
   }
 
-  def stripExtraNewLinesAndComments(input: String): String = {
+  def stripExtraNewLinesAndCommentsUsingRegexp(input: String): String = {
     extraNewLinesRegexp.replaceAllIn(commentRegexp.replaceAllIn(input, ""), "\n")
   }
+
+  private object State extends Enumeration {
+    val TEXT, SEPARATOR, SEPARATOR_WITH_NEWLINE, MAYBE_COMMENT, LINE_COMMENT, BLOCK_COMMENT,
+    BLOCK_COMMENT_WITH_NEWLINE, MAYBE_BLOCK_COMMENT_CLOSING,
+    MAYBE_BLOCK_COMMENT_WITH_NEWLINE_CLOSING = Value
+  }
+
+  def stripExtraNewLinesAndComments(input: String): String = {
```

Comment:
Could you brush up code more (e.g., keep the same code style for consistency) by referring `IntervalUtils. stringToInterval`? Probably, I think @MaxGekk could help you ;)
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala#L623

Reply:
We build a customer parser for intervals because the input is `UTF8String`. Is there any standard way to do the string replacement efficiently in the java world?

Reply:
I've brushed it up a bit. I left `if`s instead of `match/case`s where we have only 2 or less branches. Let me know if it needs more brushing.